common/src/main/java/com/netflix/conductor/common/metadata/workflow/TaskType.java

```diff
@@ -0,0 +1,46 @@
+package com.netflix.conductor.common.metadata.workflow;
+
+import com.github.vmg.protogen.annotations.ProtoEnum;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@ProtoEnum
+public enum TaskType {
+
+    SIMPLE, DYNAMIC, FORK_JOIN, FORK_JOIN_DYNAMIC, DECISION, JOIN, SUB_WORKFLOW, EVENT, WAIT, USER_DEFINED;
+
+    /**
+     * TaskType constants representing each of the possible enumeration values.
+     * Motivation: to not have any hardcoded/inline strings used in the code.
+     * Example of use: CoreModule
+     */
+    public static final String TASK_TYPE_DECISION = "DECISION";
+    public static final String TASK_TYPE_DYNAMIC = "DYNAMIC";
+    public static final String TASK_TYPE_JOIN = "JOIN";
+    public static final String TASK_TYPE_FORK_JOIN_DYNAMIC = "FORK_JOIN_DYNAMIC";
+    public static final String TASK_TYPE_EVENT = "EVENT";
+    public static final String TASK_TYPE_WAIT = "WAIT";
+    public static final String TASK_TYPE_SUB_WORKFLOW = "SUB_WORKFLOW";
+    public static final String TASK_TYPE_FORK_JOIN = "FORK_JOIN";
+    public static final String TASK_TYPE_USER_DEFINED = "USER_DEFINED";
+    public static final String TASK_TYPE_SIMPLE = "SIMPLE";
+
+    private static Set<String> systemTasks = new HashSet<>();
```

Comment:
I know this is existing code, but we should introduce a isSystemTask flag to the enum to remove this set.

Reply:
Sure, sounds something feasible for this PR 👍